netket/sampler/metropolis_hastings.py

```diff
@@ -5,10 +5,15 @@
 from .._C_netket.utils import random_engine
 from ..stats import mean as _mean
 
-from numba import jit, jitclass
-from numba import int64, float64
 from netket import random as _random
 
+from numba import jit, int64, float64
+from pkg_resources import get_distribution
+if get_distribution("numba").version < '0.49':
+    from numba import jitclass
+else:
+    from numba.experimental import jitclass
+
```

Comment:
Maybe it would be better to put this code in `sampler/__init__.py` or even `sampler/_jitclass.py` and import the name from there, so that there is no duplication and we don't have to copy-paste these lines even more if we use `jitclass` somewhere else.

Reply:
I'd put it in netket/_jitclass so that if ever used somewhere else it's accessible?